### PR TITLE
Allow embedding of forms on non-dev URLs

### PR DESCRIPTION
**Pull request by MarekBodingerBA**
Up until now the embedding was only allowed on DEV urls, it needed more robust solution as in non-dev URLs the form is first created (which yields ID) and the query param (`?externa-sluzba=true`) must be carried on to the new page. Implemented a function that does this on all three places.

### CODE REVIEW

File: next/frontend/utils/embeddedFormsHelpers.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+  return map[formDefinition.embedded]
+}
+
+export const handleEmbeddedFormRequest = (
+  formDefinition: FormDefinition,
+  context: GetServerSidePropsContext,
+): { success: false; isEmbedded: undefined } | { success: true; isEmbedded: boolean } => {
+  const isEmbeddedQueryParam =
+    context.query[EMBEDDED_FORM_QUERY_PARAM] === EMBEDDED_FORM_QUERY_PARAM_TRUE_VALUE
```
Comment (mpinter) [inline next/frontend/utils/embeddedFormsHelpers.ts:26]: the `EMBEDDED_FORM_QUERY_PARAM_TRUE_VALUE` seems a bit cumbersome

any drawbacks to just `EMBEDDED_FORM_QUERY_PARAM in context.query` and

```ts
 const isEmbeddedPostfix = isEmbedded
    ? EMBEDDED_FORM_QUERY_PARAM
    : ''
```

**Review by mpinter — APPROVED** [review]
Check comments, one more linter warning for pipelines to pass, otherwise should be good.

File: next/frontend/utils/embeddedFormsHelpers.ts (hunk 1 above)
Comment (MarekBodingerBA) [reply next/frontend/utils/embeddedFormsHelpers.ts:26]: I tried this and wanted it to be this way, but Next doesn't work with query params like this unfortunately.

File: next/frontend/utils/embeddedFormsHelpers.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+export const EMBEDDED_FORM_QUERY_PARAM = 'externa-sluzba'
+export const EMBEDDED_FORM_QUERY_PARAM_TRUE_VALUE = 'true'
+
+const getEmbeddedFormsAllowedOrigins = (formDefinition: FormDefinition) => {
+  if (!formDefinition.embedded) {
+    return null
+  }
+
+  return map[formDefinition.embedded]
```
[the remark this answers is not in the document]
Comment (MarekBodingerBA) [reply next/frontend/utils/embeddedFormsHelpers.ts:18]: I will think about it, but the `olo` value will be used for more than allowed origins.

File: next/frontend/utils/embeddedFormsHelpers.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    }
+  }
+
+  const allowedOrigins = getEmbeddedFormsAllowedOrigins(formDefinition)
+  // Form is not meant to be embedded.
+  if (!allowedOrigins) {
+    return { success: false, isEmbedded: undefined }
+  }
+
+  context.res.setHeader('Content-Security-Policy', `frame-ancestors ${allowedOrigins.join(' ')}`)
```
[the remark this answers is not in the document]
Comment (MarekBodingerBA) [reply next/frontend/utils/embeddedFormsHelpers.ts:42]: 👍

**Pull request merged**